python/tvm/topi/tensor.py

```diff
@@ -73,3 +75,341 @@ def full_like(x, fill_value):
+        input data
+
+    segment_ids : tvm.te.Tensor
+        input segment ids
+
+    num_out : int
+        number of output
+
+    Returns
+    -------
+    out : tvm.te.Tensor
+        Tensor with shape determined by the segment ids.
+    """
+
+    def _segment_max(data, segment_ids, out_buf):
+
+        ib = tir.ir_builder.create()
+        input_data = ib.buffer_ptr(data)
+        seg_ids = ib.buffer_ptr(segment_ids)
+        out = ib.buffer_ptr(out_buf)
+
+        shape = get_const_tuple(data.shape)
+        num_segment = get_const_tuple(out_buf.shape)[0]
+        inner_size = 1
+        for s in range(1, len(shape)):
+            inner_size = inner_size * shape[s]
+
+        with ib.for_range(0, num_segment) as n:
+            with ib.for_range(0, inner_size) as j:
+                out_index = n * inner_size + j
+                out[out_index] = -3.4028235e38
+
+            with ib.for_range(0, shape[0]) as k:
+                with ib.if_scope(seg_ids[k] == n):
+                    with ib.for_range(0, inner_size) as l:
+                        out_index = n * inner_size + l
+                        in_index = k * inner_size + l
+                        out[out_index] = te.max(input_data[in_index], out[out_index])
+
+        return ib.get()
+
+    assert len(segment_ids.shape) == 1
+
+    out_shape = list(get_const_tuple(data.shape))
+    out_shape[0] = num_out
+
+    out = te.extern(
+        out_shape,
+        [data, segment_ids],
+        lambda ins, outs: _segment_max(ins[0], ins[1], outs[0]),
+        dtype=data.dtype,
+    )
+
+    return out
+
+
+def segment_min(data, segment_ids, num_out):
+    """segment_min operator.
+
+    Parameters
+    ----------
+    data : tvm.te.Tensor
+        input data
+
+    segment_ids : tvm.te.Tensor
+        input segment ids
+
+    num_out : int
+        number of output
+
+    Returns
+    -------
+    out : tvm.te.Tensor
+        Tensor with shape determined by the segment ids.
+    """
+
+    def _segment_min(data, segment_ids, out_buf):
+
+        ib = tir.ir_builder.create()
+        input_data = ib.buffer_ptr(data)
+        seg_ids = ib.buffer_ptr(segment_ids)
+        out = ib.buffer_ptr(out_buf)
+
+        shape = get_const_tuple(data.shape)
+        num_segment = get_const_tuple(out_buf.shape)[0]
+        inner_size = 1
+        for s in range(1, len(shape)):
+            inner_size = inner_size * shape[s]
+
+        with ib.for_range(0, num_segment) as n:
+            with ib.for_range(0, inner_size) as j:
+                out_index = n * inner_size + j
+                out[out_index] = 3.4028235e38
+
+            with ib.for_range(0, shape[0]) as k:
+                with ib.if_scope(seg_ids[k] == n):
+                    with ib.for_range(0, inner_size) as l:
+                        out_index = n * inner_size + l
+                        in_index = k * inner_size + l
+                        out[out_index] = te.min(input_data[in_index], out[out_index])
+
+        return ib.get()
```

Comment:
It looks like these function share a common implementation. Could you write a single `_segment_op` with a string `op`  parameters and add some logic inside this function?

Reply:
👌